src/odb/src/db/dbInsertBuffer.cpp

```diff
@@ -0,0 +1,1685 @@
+  if (term->getObjectType() == dbITermObj) {
+    dbITerm* iterm = static_cast<dbITerm*>(term);
+    iterm->disconnect();  // Disconnect both flat and mod nets
+    iterm->connect(new_flat_net_);
+  } else if (term->getObjectType() == dbBTermObj) {
+    dbBTerm* bterm = static_cast<dbBTerm*>(term);
+    bterm->disconnect();
+    bterm->connect(new_flat_net_);
+  }
+}
+
+bool dbInsertBuffer::getPinLocation(dbObject* pin, int& x, int& y) const
+{
+  if (pin == nullptr) {
+    return false;
+  }
+
+  if (pin->getObjectType() == dbITermObj) {
+    dbITerm* iterm = static_cast<dbITerm*>(pin);
+    if (iterm->getAvgXY(&x, &y)) {
+      return true;
+    }
+
+    // Fallback to instance origin
+    dbInst* inst = iterm->getInst();
+    Point origin = inst->getOrigin();
+    x = origin.getX();
+    y = origin.getY();
+    return true;
+  }
+
+  if (pin->getObjectType() == dbBTermObj) {
+    dbBTerm* bterm = static_cast<dbBTerm*>(pin);
+    return bterm->getFirstPinLocation(x, y);
+  }
+
+  return false;
+}
+
+Point dbInsertBuffer::computeCentroid(
+    dbObject* drvr_pin,
+    const std::set<dbObject*>& load_pins) const
+{
+  uint64_t sum_x = 0;
+  uint64_t sum_y = 0;
+  int count = 0;
+
+  // Include driver pin
+  int x = 0, y = 0;
+  if (getPinLocation(drvr_pin, x, y)) {
+    sum_x += x;
+    sum_y += y;
+    count++;
+  }
+
+  // Include load pins
+  for (dbObject* pin : load_pins) {
+    if (getPinLocation(pin, x, y)) {
+      sum_x += x;
+      sum_y += y;
+      count++;
+    }
+  }
+
+  if (count == 0) {
+    return Point(0, 0);
+  }
+
+  return Point(static_cast<int>(sum_x / count),
+               static_cast<int>(sum_y / count));
+}
+
+dbNet* dbInsertBuffer::createNewFlatNet(std::set<dbObject*>& connected_terms)
+{
+  // Create a new net for buffering in the target module.
+  //
+  // Algorithm:
+  // - The new net will drive the connected_terms, and the original net will be
+  //   connected to the buffer terminal.
+  // - If the original net name matches a connected BTerm, rename the original
+  //   net to a unique name and assign the BTerm's name to the new net.
+  //   This ensures that the net name matches the port name for Verilog
+  //   compatibility.
+
+  std::string new_net_name = (new_net_base_name_) ? new_net_base_name_ : "net";
+  dbNameUniquifyType new_net_uniquify = uniquify_;
+
+  // Check if the net name conflicts with any port name
+  for (dbObject* obj : connected_terms) {
+    if (obj->getObjectType() != dbBTermObj) {
+      continue;
+    }
+
+    dbBTerm* bterm = static_cast<dbBTerm*>(obj);
+    std::string_view bterm_name{bterm->getConstName()};
+    if (bterm_name == block_->getBaseName(net_->getConstName())) {
+      // The original net names uses the BTerm name.
+      // Rename this net if its name is the same as a port name in loads_pins
+      std::string new_orig_net_name = block_->makeNewNetName(
+          target_module_ ? target_module_->getModInst() : nullptr,
+          "net",
+          uniquify_);
```

Comment:
If uniquify is IF_NEEDED* this may not produce a different name.  Would it be better to use ALWAYS*?